synapse/storage/_base.py

```diff
@@ -1693,3 +1695,49 @@ def db_to_json(db_content):
     except Exception:
         logging.warning("Tried to decode '%r' as JSON and failed", db_content)
         raise
+
+
+def add_in_list_sql_clause(
```

Comment:
I'm not convinced that this function adds much in the way of clarity tbh. I'd be inclined to inline it.